src/windows/scanner.cc

```diff
@@ -144,100 +144,56 @@ static drivelist::Code ScanDisks(drivelist::com::Connection *const connection,
   if (FAILED(result))
     return InterpretHRESULT(result);
 
-  drivelist::wmi::Query query(L"SELECT * FROM Win32_Volume");
-  result = query.Execute(connection);
+  std::vector<wchar_t> volumes;
+  result = drivelist::volume::GetAvailableVolumes(&volumes);
   if (FAILED(result))
     return InterpretHRESULT(result);
 
-  BSTR string;
-  ULONG number;
-
-  while (query.HasResult()) {
-    result = query.GetPropertyString(L"DriveLetter", &string);
-    if (FAILED(result))
-      return InterpretHRESULT(result);
-    if (string == NULL) {
-      result = query.SelectNext();
-      if (FAILED(result))
-        return InterpretHRESULT(result);
+  for (wchar_t volume : volumes) {
+    // We only want to consider removable or local disks in this module
+    drivelist::volume::Type volumeType = drivelist::volume::GetType(volume);
+    if (volumeType != drivelist::volume::Type::REMOVABLE_DISK &&
+        volumeType != drivelist::volume::Type::LOCAL_DISK) {
       continue;
     }
 
-    std::string path = ConvertBSTRToString(string);
-    SysFreeString(string);
-
     BOOL hasFilesystem;
-    result = drivelist::volume::HasFileSystem(path[0], &hasFilesystem);
+    result = drivelist::volume::HasFileSystem(volume, &hasFilesystem);
     if (FAILED(result))
       return InterpretHRESULT(result);
 
-    // We only want to consider removable or local disks in this module
-    drivelist::volume::Type volumeType = drivelist::volume::GetType(path[0]);
-    if (volumeType != drivelist::volume::Type::REMOVABLE_DISK &&
-        volumeType != drivelist::volume::Type::LOCAL_DISK) {
-      result = query.SelectNext();
-      if (FAILED(result))
-        return InterpretHRESULT(result);
-      continue;
-    }
-
-    result = drivelist::volume::GetDeviceNumber(path[0], &number);
+    ULONG number;
+    result = drivelist::volume::GetDeviceNumber(volume, &number);
     if (FAILED(result))
       return InterpretHRESULT(result);
 
     // Turns out a disk can be writable while its volumes can be read-only
     // and viceversa, so we need to check the writable capabilities of
     // both the disk and its volumes
     BOOL writable;
-    result = drivelist::volume::IsDiskWritable(path[0], &writable);
+    result = drivelist::volume::IsDiskWritable(volume, &writable);
     if (FAILED(result))
       return InterpretHRESULT(result);
     if (writable && hasFilesystem) {
-      result = drivelist::volume::IsVolumeWritable(path[0], &writable);
+      result = drivelist::volume::IsVolumeWritable(volume, &writable);
       if (FAILED(result))
         return InterpretHRESULT(result);
     }
 
     drivelist::mountpoint_s mountpoint;
-    mountpoint.path = path;
+    mountpoint.path = std::string(1, static_cast<char>(volume)) + ":";
     mountpoint.disk = "\\\\.\\PHYSICALDRIVE" + std::to_string(number);
     mountpoint.readonly = !writable;
-    mountpoint.system = path[0] == systemDriveLetter;
+    mountpoint.system = volume == systemDriveLetter;
```

Comment:
> Ah. To prevent confusion (like I just had!) maybe it'd better to use an attribute-name other than system here?

I guess `system` makes a lot of sense in this case, so maybe worth changing the other `system` property to `removable`?

> ...and given that you're determining here whether a disk/volume/drive is REMOVABLE_DISK or LOCAL_DISK, would it be worth storing that info on the mountpoint object too?

I'm determining this for disks, not for mountpoints, so I wouldn't be able to add that to the mountpoint object.